### ISSUE TITLE
Potential bug in accuracy computation

### ISSUE THREAD

**Issue by giorgioskij** [atticusg/CausalAbstraction#3]
Hi!
I was going through the code with a debugger to understand the experiment pipeline, and I noticed something weird. 
During computation of the accuracy score, in the method `interpret_results` of the `InterventionExperiment` class, line `498` is:

```
for example, text, raw_output in zip(labeled_data, texts, raw_outputs[0]):
```

However, at this step, in my execution:
- `labeled data` is a list of target labels of length `951`
- `texts` is a list of model outputs of length `951`
- `raw_outputs` is a list of 8 **batches** of model outputs, each containing `batch_size==128` raw model outputs (the last batch contains only 55, since `128 * 7  + 55 == 951`

![Image](https://github.com/user-attachments/assets/70f061b0-e196-4e30-b495-7345f1edd85e)
(see below for my exact code to reproduce the issue).

When `zip` receives arguments of different lengths, it only considers the first `n`, where `n` is the length of the **shortest** sequence. Since here `zip` receives `raw_outputs[0]`, which has length equal to `batch_size`, this means that only the first `batch_size` elements of `labeled_data` and `texts` are considered to compute the accuracy. 

In practice, if I am not mistaken, this would mean that the **accuracy is only computed over the first batch of data** instead of over the whole dataset. 

Is this a bug, or am I missing something? Maybe my configuration for the experiment is not correct, and this is an unwanted side effect?

Thank you for your contribution to a standardization of MechInt research!

--------
## Reproducibility
Below follows the code to reproduce my execution scenario. This comes from a notebook after condensing all cells into a single python script.
**NOTE** that, as mentioned in #2, I had to add a parameter `include_private: bool = False` to the `get_task` function to fetch only the public dataset. If you want, I can submit a quick PR to bugfix that. 

```
# Load Task
from CausalAbstraction.tasks.two_digit_addition_task.arithmetic import get_task
task = get_task(hf=True)

# Load model
import torch
from CausalAbstraction.pipeline import LMPipeline
device = "cuda" if torch.cuda.is_available() else "cpu"
model_name = "google/gemma-2-2b"
pipeline = LMPipeline(model_name,
                      max_new_tokens=1,
                      device=device,
                      dtype=torch.float16)
pipeline.tokenizer.padding_side = "left"
pipeline.max_new_tokens = 3

# Display task example
from CausalAbstraction.tasks.two_digit_addition_task.arithmetic import get_token_positions


print("COUNTERFACTUAL DATA: ")
task.display_counterfactual_data()

print("\nDATASET:")
print(next(iter(task.all_data)))

print("\nEXAMPLE INPUT / OUTPUT / GEN:")
i = task.raw_all_data["input"][0]
print(f"INPUT: {i}")

label = task.raw_all_data["label"][0]
print(f"CORRECT: {label}")

gen = pipeline.generate(i)
output = pipeline.dump(gen)
print(f"OUTPUT: {output}")

print("\nTOKEN POSITIONS: ")
token_positions = get_token_positions(pipeline, task)
print(token_positions[0].highlight_selected_token(i))
print(token_positions[1].highlight_selected_token(i))

# Filter task for agreement between prediction and causal model output
import re

def checker(output_text: str, expected: str):
    f = re.findall(r"\d+", output_text)
    if len(f) == 0:
        return False
    return int(f[0]) == int(expected)

print(f'{checker(output, label)=}')

task.filter(pipeline, checker, verbose=True, batch_size=2048)
import gc
gc.collect()
torch.cuda.empty_cache()

# Prepare experiment for Full Vector Baseline

from CausalAbstraction.experiments.LM_experiments import PatchResidualStream

config = {
    "method_name": "full_vector",
    "batch_size": 128,
}

n_layers = len(pipeline.model.model.layers)
layers = [10]

experiment = PatchResidualStream(pipeline,
                                 task,
                                 layers,
                                 token_positions,
                                 checker,
                                 config=config)

# split train, validation, test data
test_data = ["random_test", "ones_carry_test"]

# perform interventions to text experiment
raw_results = experiment.perform_interventions(test_data, verbose=True)

# interpret results
target_variables = ["ones_carry"]
results_dir = "./arithmetic_results"
processed_results = experiment.interpret_results(raw_results,
                                                 target_variables,
                                                 save_dir=results_dir)
```

**Comment by atticusg** [comment]
This was indeed a bug! Luckily, I noticed this as well during my refactoring and testing of the codebase. In the new refactored code we now have a line flattening the raw_results:

https://github.com/atticusg/CausalAbstraction/blob/431c9455717691e10179330e8f08f37dad418faa/experiments/intervention_experiment.py#L138

The ICML camera ready version of the MIB paper has results from after my bug fix as does the new arxiv of the paper.

Again, I must thank you for this thorough investigation of the code! I will try to be more quick to respond moving forward :)

**Comment by giorgioskij** [comment]
alright, perfect! I later noticed that most of the times I could easily fit in vRAM a batch size bigger than the whole test dataset, so this problem would be completely bypassed anyways in practice
